### PR TITLE
fix: html repr for xarray

### LINKED ISSUE #3568: `DataGroup._repr_html_` does not display sizes of Xarray objects
Try
```python
import scipp as sc
import xarray as xr

sc.DataGroup(a=xr.DataArray(dims=['z'], data=[1.0, 2.0, 3.0, 4.0]))
```
screenshot-2024-10-16T09:01:43

`DataGroup` handles (and displayes) the `xr.DataArray.sizes` correctly when computing its overall size. The HTML repr (in the line for `a`) however just displayes `()` as if it were 0-D. This is what `DataGroup` does for other Python objects that do not expose `sizes`, but Xarray objects do and should be handled better.

### PR DESCRIPTION
**Pull request by jokasimr**
Fixes #3568

### CODE REVIEW

**Comment by jokasimr** [comment]
Screenshot from 2024-10-21 11-45-36

File: src/scipp/visualization/formatting_datagroup_html.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 import numpy as np
 import numpy.typing as npt
+import xarray as xr
```
Comment (MridulS) [inline src/scipp/visualization/formatting_datagroup_html.py:11]: the wheel builds are failing as this assumes xarray will be available during wheel build but we only have xarray in the `extra` requirements.

Comment (MridulS) [reply src/scipp/visualization/formatting_datagroup_html.py:11]: short of adding `xarray` as a required dep (I don't think we should), we can add a try except block here:
```
try:
     import xarray as xr
     xarray_installed = True
except:
     xarray_installed = False
```

and the elif condition below could be something like `elif xarray_installed and isinstance(var, xr.DataArray)`:

Comment (jokasimr) [reply src/scipp/visualization/formatting_datagroup_html.py]: Makes sense! If xarray is not installed we won't see it in the DataGroup anyway. I'll do that

File: src/scipp/visualization/formatting_datagroup_html.py [hunk 2]
```diff
@@ -86,6 +93,9 @@ def _summarize_atomic_variable(var: Any, name: str, depth: int = 0) -> str:
     if isinstance(var, Dataset | DataArray | Variable):
         parent_obj_str = "scipp"
         shape_repr = _format_shape(var)
+    if xarray_installed and isinstance(var, xr.DataArray | xr.Dataset):
+        parent_obj_str = "xarray"
+        shape_repr = _format_shape(var)
```
Comment (SimonHeybrock) [inline src/scipp/visualization/formatting_datagroup_html.py:98]: Can we avoid importing Xarray entirely? What we mainly want is check for the presence of a `sizes` attribute, just like `DataGroup.__getitem__`. And the parent module could be extracted from the qualified name?

**Review by SimonHeybrock — CHANGES_REQUESTED** [review]

File: src/scipp/visualization/formatting_datagroup_html.py (hunk 2 above)
Comment (jokasimr) [reply src/scipp/visualization/formatting_datagroup_html.py]: Yes that's true, fixed now

File: src/scipp/visualization/formatting_datagroup_html.py [hunk 3]
```diff
@@ -86,6 +86,15 @@ def _summarize_atomic_variable(var: Any, name: str, depth: int = 0) -> str:
...
         shape_repr = _format_shape(var)
+    if hasattr(var, '__module__'):
+        parent_obj_str = var.__module__.partition('.')[0]
+    if hasattr(var, 'sizes') and isinstance(var.sizes, Mapping):
+        try:
+            shape_repr = _format_shape(var)
+        except Exception:  # noqa: S110
+            # We don't want the rendering to fail because some
+            # object did not behave the way we expected here
+            pass
```
Comment (SimonHeybrock) [inline src/scipp/visualization/formatting_datagroup_html.py:97]: Either you meant to use `elif` somewhere, or the branch for Scipp can be removed?

**Comment by SimonHeybrock** [comment]
Have you verified that the latest version still displays Scipp objects correctly?

**Comment by jokasimr** [comment]
> Have you verified that the latest version still displays Scipp objects correctly?

Yes :+1:

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**